ql/src/java/org/apache/hadoop/hive/ql/exec/Task.java

```diff
@@ -176,7 +176,8 @@ protected void setState(TaskState state) {
 
   protected Hive getHive() {
     try {
-      return Hive.getWithFastCheck(conf);
+      //Hive.getWithFastCheck(conf) will cause leaked tcp link to metastore
```

Comment:
they end up calling the same getInternal method. The only diff is in the 3rd parameter, isFastCheck. Is the result of isCompatible different for these? Perhaps getInternal should close the db if it's not compatible instead.